rpcs3/Emu/System.cpp

```diff
@@ -101,6 +101,24 @@ void fmt_class_string<game_boot_result>::format(std::string& out, u64 arg)
 	});
 }
 
+template<>
+void fmt_class_string<cfg_mode>::format(std::string& out, u64 arg)
```

Comment:
What's wrong with keeping all logic in m_config_path? Because std::string is usually defined everywhere it can be used in headers which do not include System.h very cleany.
Formatting is implicitly declated and it is one less arhument passed to BootGame.

Reply:
On the flip side, this is cleaner. Encoding settings in a config path and having special interpretation of strings is not the best approach. This approach is more explicit about what is going on which is easier to maintain when the team inevitably changes in the future.